MapboxNavigation/RouteMapViewController.swift

```diff
@@ -476,9 +486,68 @@ class RouteMapViewController: UIViewController {
             self.laneViewsContainerView.isHidden = true
         }, completion: nil)
     }
+    //MARK: - End Of Route
+    
+    func showEndOfRoute(duration: TimeInterval = 0.3, completion: ((Bool) -> Void)? = nil) {
+        self.view.layoutIfNeeded() //flush layout queue
+        endOfRouteViewController?.destination = route.legs.last?.destination
+        
+        self.endOfRouteContainer.isHidden = false
+        self.endOfRouteHide.isActive = false
+        self.endOfRouteShow.isActive = true
+        self.bannerHide.isActive = true
+        self.bannerShow.isActive = false
+        self.bannerContainerShow.isActive = false
+        
+        mapView.enableFrameByFrameCourseViewTracking(for: duration)
+        mapView.setNeedsUpdateConstraints()
+        
+        let layout = view.layoutIfNeeded
+        let noAnimation = { layout(); completion?(true) }
+
+        guard duration > 0.0 else { return noAnimation() }
+        UIView.animate(withDuration: duration, delay: 0.0, options: [.curveLinear], animations: layout, completion: completion)
+    }
+    
+    func hideEndOfRoute(duration: TimeInterval = 1.0, completion: ((Bool) -> Void)? = nil) {
+        self.view.layoutIfNeeded() //flush layout queue
+        
+        self.endOfRouteHide.isActive = true
+        self.endOfRouteShow.isActive = false
+        self.view.clipsToBounds = true
+        
+        //NOTE: This isn't perfect, but it's better then using the pattern in showEndOfRoute(duration:completion:).
+        mapView.contentInset.top = instructionsBannerContainerView.bounds.height
+        mapView.contentInset.bottom = bottomBannerView.bounds.height
+        
+        mapView.enableFrameByFrameCourseViewTracking(for: duration)
+        mapView.setNeedsUpdateConstraints()
+
+        let layout = view.layoutIfNeeded
+        let complete: (Bool) -> Void = { self.endOfRouteContainer.isHidden = true; completion?($0)}
+        let noAnimation = { layout(); complete(true) }
+
+        guard duration > 0.0 else { return noAnimation() }
+       UIView.animate(withDuration: duration, delay: 0.0, options: [.curveLinear], animations: layout, completion: complete)
+    }
+    
+    override func prepare(for segue: UIStoryboardSegue, sender: Any?) {
+        guard let endOfRouteVC = segue.destination as? EndOfRouteViewController else { return }
+        
+        endOfRouteVC.dismiss = { (stars, comment) in
```

Comment:
Might be mistaken but this looks like a ref cycle. It's not a ref cycle on this line already since `endOfRouteVC` is a scope variable but further down, `endOfRouteViewController = endOfRouteVC` <- strong ref, and using self inside the closure would trigger a ref cycle. We should use a weak self here.

Reply:
- [ ] remove retain cycle